iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/RegionMigrateService.java

```diff
@@ -192,7 +192,7 @@ private boolean addToTaskResultMap(long taskId) {
     if (taskResultMap.containsKey(taskId)) {
       return false;
     }
-    taskResultMap.put(taskId, unfinishedResult);
+    taskResultMap.putIfAbsent(taskId, unfinishedResult);
```

Comment:
putifAbsent

Reply:
you can always learn something from Chief Tan !